src/wh_server.c

```diff
@@ -84,16 +85,73 @@ int wh_Server_Cleanup(whServerContext* server)
 
     memset(server, 0, sizeof(*server));
 
-    return 0;
+    return WH_ERROR_OK;
 }
 
+int wh_Server_SetConnected(whServerContext *server, int connected)
+{
+    if (server == NULL) {
+        return WH_ERROR_BADARGS;
+    }
+
+    server->connected = connected;
+    return WH_ERROR_OK;
+}
+
+int wh_Server_SetConnectedCb(void* s, int connected)
```

Comment:
I had a common grumble on this.  The problem is that shared-memory platforms need to use some mechanism outside of the transport to indicate connected/disconnected because it's not necessary safe to use the shared-memory before it is configured.  So, the server/client context is the one that needs to maintain this upper-level connectedness, but the lower-level may (in some circumstances) be in the best position to detect connects and disconnects.  Notice in the Posix TCP transport implementation, we ALWAYS set connected initially because the connection sequence is handled during the Recv function, which will only be called if the upper level thinks its connected.  Chicken/egg.  The disconnect is automatically set by the send/recv functions when it notices the socket is no longer connected.  I did not wire in the client side yet as I'm not convinced we actually need this on the client side because the server won't disconnect on purpose and the client is the one the instantiates the connection/shared-memory so there's never an issue here.  

As for multiple connections, recall that each client/server context handles only 1 connection at a time.  The server app instantiations multiple server contexts, which share NVM and other singleton components.